### PR TITLE
Fix to_time_series_item in splitter.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lostella** [comment]
Maybe we can add a small test for these components?

**Comment by dcmaddix** [comment]
> Maybe we can add a small test for these components?

Yes, let's add a smoke test for this.  I can add a smoke test for the failing case we saw.

**Comment by dcmaddix** [comment]
Great, this should fix the current error:
 ```
ValidationError: 2 validation errors for TimeSeriesItem
feat_dynamic_real -> 0
  value is not a valid list (type=type_error.list)
feat_dynamic_real -> 1
  value is not a valid list (type=type_error.list)
```
on line `--> 109             feat_dynamic_real=[real.values for real in self.feat_dynamic_real],`

where  feat_dynamic_real is of type `List[np.ndarray]` but the `TimeSeriesItem` requires type `List[List[float]]`.

**Comment by jaheba** [comment]
I've added a quick test. Without the changes it fails.

File: test/dataset/split/test_split.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    sl = TimeSeriesSlice(
+        target=make_series(range(100)),
+        item="",
+        feat_static_cat=[1, 2, 3],
+        feat_static_real=[0.1, 0.2, 0.3],
+        feat_dynamic_cat=[make_series(range(100))],
+        feat_dynamic_real=[make_series(range(100))],
+    )
+
+    sl.to_time_series_item()
```
Comment (dcmaddix) [inline test/dataset/split/test_split.py:35]: Looks good, otherwise this line would fail.  Thanks!

**Pull request merged**